src/plugins/sync-plugin.js

```diff
@@ -178,6 +180,39 @@ export const getRelativeSelection = (pmbinding, state) => ({
   head: absolutePositionToRelativePosition(state.selection.head, pmbinding.type, pmbinding.mapping)
 })
 
+const getElementFromTextNode = node => {
+  if (dom.checkNodeType(node, dom.ELEMENT_NODE)) {
+    return node
+  }
+  return node.parentElement
+}
+
+const isDomSelectionInView = () => {
+  const selection = window.getSelection()
+  const anchorElement = getElementFromTextNode(selection.anchorNode)
```

Comment:
I think it would be easier to just compute the bounding rect of the selection range directly. This approach is more granular and will also account for the case that there is a huge paragraph with a selection:

```js
// const range = selection.getRangeAt(0) // this is an alternative approach to get the range, but it is marked as experimental on mdn
// This is why we create a selection range manually:
const range = view._root.createRange()
range.setStart(selection.anchorNode, selection.anchorOffset)
range.setEnd(selection.focusNode, selection.focusOffset)
const rect = range.getBoundingClientRect()
```